Version2/index.js

```diff
@@ -0,0 +1,128 @@
+});
+
+document.addEventListener("mousemove", (e) => {
+  const x = e.pageX;
+  const y = e.pageY;
+  const buttonBox = button.getBoundingClientRect();
+  const horizontalDistanceFrom = distanceFromCenter(
+    buttonBox.x,
+    x,
+    buttonBox.width
+  );
+  const verticalDistanceFrom = distanceFromCenter(
+    buttonBox.y,
+    y,
+    buttonBox.height
+  );
+
+  mouse_from_right =
+    horizontalDistanceFrom >= -145 &&
+    horizontalDistanceFrom <= -45 &&
+    verticalDistanceFrom <= 20 &&
+    verticalDistanceFrom >= -25;
+  if (mouse_from_right) {
+    console.log("Rechts");
+    set_button_position(buttonBox.x - 100 * 6);
+  }
+
+  mouse_from_left =
+    horizontalDistanceFrom <= 145 &&
+    horizontalDistanceFrom >= 45 &&
+    verticalDistanceFrom <= 20 &&
+    verticalDistanceFrom >= -25;
+  if (mouse_from_left) {
+    console.log("Links");
+    set_button_position(buttonBox.x - 100 * 5);
+  }
+
+  mouse_from_top =
+    horizontalDistanceFrom >= -70 &&
+    horizontalDistanceFrom <= 67 &&
+    verticalDistanceFrom <= 72 &&
+    verticalDistanceFrom >= 18;
+  if (mouse_from_top) {
+    console.log("Oben");
+    set_button_position(0, buttonBox.y - 100 * 3);
+  }
+
+  mouse_from_bottom =
+    horizontalDistanceFrom >= -70 &&
+    horizontalDistanceFrom <= 67 &&
+    verticalDistanceFrom >= -72 &&
+    verticalDistanceFrom <= -22;
+  if (mouse_from_bottom) {
+    set_button_position(0, buttonBox.y - 100 * 4);
+  }
+
+  mouse_from_top_right =
+    horizontalDistanceFrom <= -70 &&
+    horizontalDistanceFrom >= -145 &&
+    verticalDistanceFrom <= 70 &&
+    verticalDistanceFrom >= 20;
+  if (mouse_from_top_right) {
+    set_button_position(buttonBox.x - 100 * 6, buttonBox.y - 100 * 3);
+  }
+
+  mouse_from_bottom_right =
+    horizontalDistanceFrom <= -70 &&
+    horizontalDistanceFrom >= -145 &&
+    verticalDistanceFrom >= -70 &&
+    verticalDistanceFrom <= -25;
+  if (mouse_from_bottom_right) {
+    set_button_position(buttonBox.x - 100 * 6, buttonBox.y - 100 * 4);
+  }
+
+  mouse_from_bottom_left =
+    horizontalDistanceFrom >= 66 &&
+    horizontalDistanceFrom <= 144 &&
+    verticalDistanceFrom >= -70 &&
+    verticalDistanceFrom <= -25;
+  if (mouse_from_bottom_left) {
+    set_button_position(buttonBox.x - 100 * 5, buttonBox.y - 100 * 4);
+  }
+
+  mouse_from_top_left =
+    horizontalDistanceFrom >= 66 &&
+    horizontalDistanceFrom <= 144 &&
+    verticalDistanceFrom <= 70 &&
+    verticalDistanceFrom >= 20;
+  if (mouse_from_top_left) {
+    set_button_position(buttonBox.x - 100 * 5, buttonBox.y - 100 * 3);
+  }
+
+  let xborder = button.getBoundingClientRect().x;
+  let yborder = button.getBoundingClientRect().y;
+
+  if (xborder <= 0 || xborder >= 1660) {
+    button.style.left = "250px";
+    button.style.top = "70px";
+  }
+
+  console.log(yborder);
+
+  if (yborder <= 7 || yborder >= 900) {
+    button.style.left = "250px";
+    button.style.top = "70px";
+  }
+});
+
+function set_button_position(left, top) {
```

Comment:
```suggestion
function setButtonPosition(left, top) {
```

Sometimes you do it right, sometimes not.